src/main/resources/fileTemplates/j2ee/velocity/Velocity Submodule build.gradle.ft

```diff
@@ -18,12 +17,11 @@ shadowJar {
     }
 }
 
-import org.apache.tools.ant.filters.ReplaceTokens
-
 task processSources(type: Sync) {
     from sourceSets.main.java.srcDirs
-    inputs.property 'version', version
-    filter ReplaceTokens, tokens: [version: version]
+    def props = [version: project.version]
+    inputs.properties props
+    expand props
```

Comment:
Comment about resource filtering applies here too.

The way this is set up for source filtering is a bit strange too -- it seems like it would lead to compile errors not linking to the correct location in the IDE.

What I usually do is generate a separate package-private source file with constants, and register that as a synchronization task.

See an example [in VanillaGradle](https://github.com/SpongePowered/VanillaGradle/blob/7548833f611dc9b856197e4a3855865918c69d37/build.gradle.kts#L102), which processes https://github.com/SpongePowered/VanillaGradle/blob/7548833f611dc9b856197e4a3855865918c69d37/src/main/templates/org/spongepowered/gradle/vanilla/internal/BuildVersions.java#L35

Reply:
I have to admit this is a tricky one because we use velocity's AP to generate the plugin manifest, and it gets the version from the plugin annotation (like with SpongeAPI 7 plugins.)

Maybe we can expand the generated manifest instead of the source file ?